planning/behavior_path_planner/src/behavior_tree_manager.cpp

```diff
@@ -27,6 +27,43 @@
 
 namespace behavior_path_planner
 {
+bool isInLanelets(const lanelet::ConstLanelets & lanes, const Pose & pose)
+{
+  for (const auto & lane : lanes) {
+    if (lanelet::utils::isInLanelet(pose, lane)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+PathWithLaneId createGoalAroundPath(
+  const std::shared_ptr<RouteHandler> & route_handler,
+  const std::optional<PoseWithUuidStamped> & modified_goal)
+{
+  const Pose goal_pose = modified_goal ? modified_goal->pose : route_handler->getGoalPose();
+  const auto shoulder_lanes = route_handler->getShoulderLanelets();
+
+  lanelet::ConstLanelet goal_lane;
+  const bool is_failed_getting_lanelet = std::invoke([&]() {
+    if (isInLanelets(shoulder_lanes, goal_pose)) {
+      return !lanelet::utils::query::getClosestLanelet(shoulder_lanes, goal_pose, &goal_lane);
+    }
+    return !route_handler->getGoalLanelet(&goal_lane);
+  });
+  if (is_failed_getting_lanelet) {
+    PathWithLaneId path{};
```

Comment:
the empty path may cause the node to die.
it's better to create the path from self_pose and offset pose like https://github.com/autowarefoundation/autoware.universe/blob/main/planning/behavior_path_planner/src/scene_module/pull_out/pull_out_module.cpp#L541-L572.
how about moving this function to utils and using it?

Reply:
↑ this function is for shoulder_lanes, so little tedious to make common.

Reply:
It seems ok for me!